manifests/charts/istio-cni/templates/daemonset.yaml

```diff
@@ -89,18 +89,14 @@ spec:
               path: /readyz
               port: 8000
           securityContext:
-            privileged: true # always requires privilege to be useful (install node plugin, etc)
+            privileged: false
             runAsGroup: 0
             runAsUser: 0
             runAsNonRoot: false
             # Both ambient and sidecar repair mode require elevated node privileges to function.
-            # But we don't need _everything_ in `privileged`, so drop+readd capabilities based on feature.
-            # privileged is redundant with CAP_SYS_ADMIN
-            # since it's redundant, hardcode it to `true`, then manually drop ALL + readd granular
-            # capabilities we actually require
+            # But we don't need _everything_ in `privileged`, so explicitly set it to false and
+            # add capabilities based on feature.
             capabilities:
-              drop:
-              - ALL
```

Comment:
Why drop this? If we do not drop ALL then add back only what we need, I think the net effect is we have more perms by default?

(granted, since we still have CAP_SYS_ADMIN, it doesn't matter)

Reply:
I don't think CAP_SYS_ADMIN is the same as privileged. Setting `securityContext.privileged=true`, then dropping all capabilities, and adding back NET_RAW, NET_ADMIN and SYS_ADMIN is not the same thing as setting `securityContext.privileged=false` and adding NET_RAW, NET_ADMIN and SYS_ADMIN. `securityContext.privileged=true` is more encompassing that CAP_SYS_ADMIN.

For example, from the failed integ-cni_istio tests, setting NET_RAW, NET_ADMIN and SYS_ADMIN was not sufficient for the repair controller. I needed to additionally set CAP_SYS_PTRACE to allow describing the pods net ns.

Reply:
I am open to keeping the `securityContext.privileged` true by default but I'd like to give users the option to set `securityContext.privileged` to false if they need to do so for compliance reasons.

It is also concerning that setting `securityContext.privileged=true` means there are additional privileged beyond the capabilities we add after dropping all.

Reply:
> I am open to keeping the `securityContext.privileged` true by default but I'd like to give users the option to set `securityContext.privileged` to false if they need to do so for compliance reasons.

If setting `securityContext.privileged=false`
- Doesn't make the pod non-privileged (it is still running as root, has total node privileges, and can grant itself any permissions)
- Still (because of the above) can't allow us to move out of the K8S `Privileged` PodSecurityStandard

then what would be the compliance reasons? It's like putting a sign on a door that says "locked", but not locking the door.

We did this before, and all it did was hopelessly confuse and mislead users. "I set `privileged` to false, so `istio-cni` isn't privileged, right"? -> no, it's still completely privileged, and runs as root on your node. There is no environment where `securityContext.privileged` is not an acceptable security risk, but running as root with CAP_SYS_ADMIN is. There are just people that misunderstand what `privileged` means.

> It is also concerning that setting `securityContext.privileged=true` means there are additional privileged beyond the capabilities we add after dropping all.

Yep, and that's a good catch. It looks like that's something in sidecar repair we missed. I am very much in favor of explicitly listing granular permissions and not relying on ~whatever mishmash of `CAP_XX` privileges `securityContext.privileged` happens to grant - it's a bad API and a vestige of the Docker days.

Reply:
To be clear, I totally agree that setting `securityContext.privileged=false` does not mean that the pod isn't privileged. And I agree that users not wanting `securityContext.privileged=true` for compliance reasons but are okay with CAP_SYS_ADMIN means they either want to check a compliance box or they misunderstand what privileged means. In this scenario I think both reasoning are valid given top down compliance requirements and the complexed of the API.

Reply:
> Yep, and that's a good catch. It looks like that's something in sidecar repair we missed. I am very much in favor of explicitly listing granular permissions under securityContext.capabilities and not relying on ~whatever mishmash of CAP_XX privileges securityContext.privileged happens to grant under whatever container runtime you happen to be using - it's a bad API and a vestige of the Docker days.

My concern is we'll only catch these instances when `securityContext.privileged` is `false`.

Reply:
Is it preferable to explicitly enumerate capabilities and guarantee there are no hidden privileges while potentially introducing confusion by setting `securityContext.privileged` to `false` OR is better to set `securityContext.privileged` to `true` and clearly communicate to the user that the container has privileged capabilities which potentially allowing unknown container privileges?

Reply:
> Still (because of the above) can't allow us to move out of the K8S Privileged PodSecurityStandard

Could you explain more about this? Not super familiar with PSS

Reply:
Yeah - if you look at [the doc ](https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline)it has better detail than I can give, but roughly:

1. There are (currently) three pod profiles, ranging from least-privileged to most: Restricted, Baseline, Privileged

2. Those three profiles are defined according to what the pod may or may not declare as privileges for itself. .e.g for a pod to be in Baseline or Restricted, among other things

- permittted `securityContext.capabilities.add` capabilities are allowlisted (i.e. no `CAP_SYS_ADMIN`)
- no host path volumes
- etc

Basically it's an attempt to fix the bad/insufficiently granular API that is `securityContext.privileged: <YES/NO>` - any pod that can't meet Restricted (the most locked down) gets bumped up to Baseline (which is the explicitly recommended default at this time for all pods), and anything that can't meet Baseline is Privileged.

Once your pod is in the PSS Privileged category, it's just Privileged. There's no additional granularity, or Semi-Privileged profile (if they add something like that, I'd be happy to put us in the lowest one we can fit into).